### PR TITLE
Node remove

### PR DESCRIPTION
**Pull request by johnsonjew**

### CODE REVIEW

File: bst.py [hunk 1]
```diff
@@ -93,6 +93,88 @@ def breadth_first(self):
...
+        q = deque([self])
+        while q:
+            node = q.popleft()
+            yield node
+            if node.left:
+                q.append(node.left)
+            if node.right:
+                q.append(node.right)
+
+    def delete(self, val):
```
Comment (Jakeand3rson) [inline bst.py:106]: Holy giant function batman. This needs to be broken up yes?

File: test_bst.py [hunk 2]
```diff
@@ -78,3 +88,43 @@ def test_post_order(full_tree):
...
+    assert list(g) == [1, 5, 14, 15, 16]
+    assert full_tree.contains(7) is False
+
+
+def test_delete_root_node(full_tree):
+    full_tree.delete(15)
+    assert full_tree.size() == 5
+    g = full_tree.in_order()
+    assert list(g) == [1, 5, 7, 14, 16]
+    assert full_tree.contains(15) is False
```
Comment (Jakeand3rson) [inline test_bst.py:130]: These tests are good... But they are all just duplicates of themselves. They aren't really testing anything new.. I suggest scrapping all but one. And add tests that check for edge cases, try and break it, and try and touch different angles of the function.
